sdk/core/System.ClientModel/src/Message/PipelineMessage.cs

```diff
@@ -32,6 +32,21 @@ protected internal PipelineMessage(PipelineRequest request)
         protected internal set;
     }
 
+    public PipelineResponse? ExtractResponse()
+    {
+        PipelineResponse? response = Response;
+        Response = null;
+        return response;
+    }
+
+    internal void AssertResponse()
```

Comment:
do we still need this?

Reply:
This is used elsewhere in the ClientModel code - perhaps we should review these instances as well, but maybe separately?

Reply:
This is called internally by the response classifiers and the response buffering logic.  I added this so that we could throw a consistent exception when we encountered this in places the response was expected to be set.  Let me know if you'd rather handle this differently.